azext_iot/tests/dps/test_iot_dps_int.py

```diff
@@ -5,779 +5,763 @@
 
-    @pytest.fixture(scope='class', autouse=True)
-    def tearDownSuite(self):
-        yield None
-        if not settings.env.azext_iot_testhub:
-            hub_attributes = self.cmd("iot hub show -n {} -g {}".format(hub, rg)).get_output_in_json()
+            self.cmd(
+                self.set_cmd_auth_type(
+                    "iot dps enrollment update -g {} --dps-name {} --enrollment-id {}"
+                    " --provisioning-status {} --etag {} --rc".format(
+                        self.entity_rg, self.entity_dps_name, enrollment_id, EntityStatusType.disabled.value, etag
+                    ),
+                    auth_type=auth_phase
+                ),
+                checks=[
+                    self.check("attestation.type", attestation_type),
+                    self.check("registrationId", enrollment_id),
+                    self.check("provisioningStatus", EntityStatusType.disabled.value),
+                    self.check("deviceId", device_id),
+                    self.check("allocationPolicy", AllocationType.hashed.value),
+                    self.check("iotHubs", self.hub_host_name.split()),
+                    self.exists("initialTwin.tags"),
+                    self.exists("initialTwin.properties.desired"),
+                    self.check("attestation.type.x509.clientCertificates.primary", None),
+                ],
+            )
 
             self.cmd(
-                "iot dps linked-hub delete --dps-name {} --linked-hub {} --resource-group {}".format(
-                    dps, hub_attributes["properties"]["hostName"], rg
-                )
+                self.set_cmd_auth_type(
+                    "iot dps enrollment delete -g {} --dps-name {} --enrollment-id {}".format(
+                        self.entity_rg, self.entity_dps_name, enrollment_id
+                    ),
+                    auth_type=auth_phase
+                ),
             )
+
+    def test_dps_enrollment_symmetrickey_lifecycle(self):
+        attestation_type = AttestationType.symmetricKey.value
+        for auth_phase in DATAPLANE_AUTH_TYPES:
+            enrollment_id, enrollment_id2 = self.generate_enrollment_names(count=2)
+            primary_key = generate_key()
+            secondary_key = generate_key()
+            device_id = self.generate_enrollment_names()[0]
+
+            etag = self.cmd(
+                self.set_cmd_auth_type(
+                    "iot dps enrollment create --enrollment-id {} --attestation-type {}"
+                    " -g {} --dps-name {} --pk {} --sk {}"
+                    " --provisioning-status {} --device-id {}"
+                    " --initial-twin-tags {} --initial-twin-properties {}"
+                    " --allocation-policy {} --rp {} --iot-hubs {} --edge-enabled".format(
+                        enrollment_id,
+                        attestation_type,
+                        self.entity_rg,
+                        self.entity_dps_name,
+                        primary_key,
```

Comment:
What about an additional test where the primary and secondary keys are generated by the service rather than user provided.

Reply:
There is a second enrollment created that uses the service to generate the keys ~100 lines down. Going to update the enrollment group tests to have user provided keys and service generated keys